File: models/Order.js
// models/Order.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

// Definir el esquema de pedido
const orderSchema = new Schema(
  {
    usuario: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    productos: [
      {
        productoId: {
          type: mongoose.Schema.Types.ObjectId,
          ref: 'Product',
          required: true,
        },
        nombre: { type: String, required: true },
        precio: { type: Number, required: true },
        cantidad: { type: Number, required: true },
        subtotal: { type: Number, required: true },
      },
    ],
    total: { type: Number, required: true },
    estado: { type: String, default: 'Pendiente' },
    direccionDestinatario: { type: String, required: true },
    nombreDestinatario: { type: String, required: true },
    telefonoDestinatario: { type: String, required: true },
    fecha: { type: Date, default: Date.now },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Order', orderSchema);

File: app/src/main/java/com/example/chocopop/Order.kt
package com.example.chocopop

data class ProductOrder(
    val productoId: String,  // ID del producto
    val nombre: String,      // Nombre del producto
    val precio: Double,      // Precio del producto
    val cantidad: Int,       // Cantidad del producto
    val subtotal: Double     // Subtotal calculado
)

data class Order(
    val usuario: String,                      // ID del usuario que realizó el pedido
    val productos: List<ProductOrder>,        // Lista de productos en el pedido
    val total: Double,                        // Total del pedido
    val fecha: String,                        // Fecha en que se realiza el pedido
    val estado: String = "Pendiente",         // Estado del pedido, por defecto "Pendiente"
    val direccionDestinatario: String,        // Dirección de entrega
    val nombreDestinatario: String,           // Nombre del destinatario
    val telefonoDestinatario: String          // Teléfono del destinatario
)

data class OrderResponse(
    val success: Boolean,
    val message: String
)

